app/javascript/components/visualization/ScatterPlot.js

```diff
@@ -134,14 +134,16 @@ function RawScatterPlot({
     }
   }, [dimensions.width, dimensions.height])
 
+
   useEffect(() => {
     $(`#${graphElementId}`).on('plotly_selected', plotPointsSelected)
     $(`#${graphElementId}`).on('plotly_legendclick', logLegendClick)
     $(`#${graphElementId}`).on('plotly_legenddoubleclick', logLegendDoubleClick)
     return () => {
-      $(`#${graphElementId}`).off('plotly_selected', plotPointsSelected)
-      $(`#${graphElementId}`).off('plotly_legendclick', logLegendClick)
-      $(`#${graphElementId}`).off('plotly_legenddoubleclick', logLegendDoubleClick)
+      $(`#${graphElementId}`).off('plotly_selected')
+      $(`#${graphElementId}`).off('plotly_legendclick')
+      $(`#${graphElementId}`).off('plotly_legenddoubleclick')
+      Plotly.purge(graphElementId)
```

Comment:
```suggestion
      Plotly.purge(graphElementId) // Needed for handling expired WebGL contexts
```

Or a better explanation for maintainers.

Reply:
The purge actually doesn't have any impact on the webgl context (those get cleared up whenever Chrome feels like once the element has been removed) -- this was just added as general good hygiene in case there are other plotly resources associated with the graph